From 21810f1b47cb55a289c639214a383b415edbb603 Mon Sep 17 00:00:00 2001
From: Rinne Zhang <RiZhang@sierrawireless.com>
Date: Tue, 9 Oct 2018 10:31:43 +0800
Subject: [PATCH] QTI9X1555-1313 - Do not change ADB device ID

Issue Description:
ADB device ID is changed after we have restarted the ADB
service or unplugged and re-plugged USB cable.

Root Cause:
USB serial number is changed by adbd daemon using value
saved in /etc/adb_devid after device is enumerated by
host using the serial number defined in android.c. New
serial number takes effect after device is re-enumerated.

Solution:
Disable regeneration of the USB Serial Number and ADB
Device ID in adbd to avoid changes during repeated USB
enumerations without device power cycling.

Resolves: QTI9X1555-1313
Signed-off-by: Rinne Zhang <RiZhang@sierrawireless.com>
---
 adb/start_adbd | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/adb/start_adbd b/adb/start_adbd
index c4b1febc5..43e09a56c 100644
--- a/adb/start_adbd
+++ b/adb/start_adbd
@@ -40,13 +40,15 @@ case "$1" in
         echo -n "Starting adbd: "
         start-stop-daemon -S -b -a /sbin/adbd

-        if [ -e "/etc/adb_devid" ]
-        then
-                cat /etc/adb_devid > /sys/class/android_usb/f_accessory/device/iSerial
-        else
-                cat /proc/sys/kernel/random/boot_id | cut -d- -f5 > /sys/class/android_usb/f_accessory/device/iSerial
-                cat /sys/class/android_usb/f_accessory/device/iSerial > /etc/adb_devid
-        fi
+        # Disable regeneration of the USB Serial Number and ADB Device ID to avoid changes
+        # during repeated USB enumerations without device power cycling
+        #if [ -e "/etc/adb_devid" ]
+        #then
+        #        cat /etc/adb_devid > /sys/class/android_usb/f_accessory/device/iSerial
+        #else
+        #        cat /proc/sys/kernel/random/boot_id | cut -d- -f5 > /sys/class/android_usb/f_accessory/device/iSerial
+        #        cat /sys/class/android_usb/f_accessory/device/iSerial > /etc/adb_devid
+        #fi

         echo "done"
         ;;
--
2.18.0

